src/devices/cpu/m6809/6x09dasm.cpp

```diff
@@ -57,42 +57,60 @@ const m6x09_base_disassembler::opcodeinfo *m6x09_base_disassembler::fetch_opcode
 {
 	uint16_t page = 0;
 	const opcodeinfo *op = nullptr;
+
 	while(!op)
 	{
 		// retrieve the opcode
 		uint16_t opcode = page | opcodes.r8(p++);
+		m6x09_instruction_level level = m_level;
 
 		// perform the lookup
 		auto iter = std::find_if(
 			m_opcodes.begin(),
 			m_opcodes.end(),
-			[opcode](const opcodeinfo &info) { return info.opcode() == opcode; });
+			[opcode,level](const opcodeinfo &info) { return (info.opcode() == opcode) && (info.level() <= level); });
 
 		// did we find something?
 		if (iter == m_opcodes.end())
-			return nullptr;
-
-		// was this a $10 or $11 page?
-		switch (iter->mode())
 		{
-		case PG1:
-			page = 0x1000;
-			break;
-
-		case PG2:
-			page = 0x1100;
-			break;
+			// on the 6809 an unimplemented page 2 or 3 opcodes fall thru to the first page.
+			if ( (m_level < HD6309_EXCLUSIVE) && (page != 0))
+			{
+				// backup the opcode pointer and disassemble the bare opcode.
+				--p;
+				page = 0;
+			}
```

Comment:
This will give no indication in the disassembly that the instruction has a technically invalid opcode prefix.  Maybe you should disassemble the prefix as a `.db` directive or something?

Also, is it really necessary to misspell “through”?

Reply:
I thought of this. I rejected it because the first page code is what is significant.
So your suggestion would produce disassembly like this:
```
0: 10 	           FCB $10
1: 11 83 55 55     CMPD #$5555
5: 11 83 55 55     CMPU #$5555
```
That makes it look like the same sequence of bytes can disassemble to different opcodes.
I like better how the following looks:
```
0: 10 11 83 55 55  CMPD   #$5555
5: 11 83 55 55     CMPU   #$5555
```
I also thought of embedding ignored page opcodes into the mnemonic:
```
0: 10 11 10 11 83 55 55  CMPD-323 #$5555
```
But it didn't seem necessary.